libs/aws/langchain_aws/chat_models/bedrock.py

```diff
@@ -260,14 +261,76 @@ def _format_anthropic_messages(
                         f"Content items must be str or dict, instead was: {type(item)}"
                     )
         else:
-            content = message.content
+            content = messages.content
 
         formatted_messages.append(
             {
                 "role": role,
                 "content": content,
             }
         )
+    else:
+        for i, message in enumerate(messages):
+            if message.type == "system":
+                if i != 0:
+                    raise ValueError(
+                        "System message must be at beginning of message list."
+                    )
+                if not isinstance(message.content, str):
+                    raise ValueError(
+                        "System message must be a string, "
+                        f"instead was: {type(message.content)}"
+                    )
+                system = message.content
+                continue
+
+            role = _message_type_lookups[message.type]
+            message_content: Union[str, List[Dict]]
+
+            if not isinstance(message.content, str):
+                # parse as dict
+                assert isinstance(
+                    message.content, list
+                ), "Anthropic message content must be str or list of dicts"
+
+                # populate content
+                message_content = []
+                for item in message.content:
+                    if isinstance(item, str):
+                        message_content.append(
+                            {
+                                "type": "text",
+                                "text": item,
+                            }
+                        )
+                    elif isinstance(item, dict):
+                        if "type" not in item:
+                            raise ValueError("Dict content item must have a type key")
+                        if item["type"] == "image_url":
+                            # convert format
+                            source = _format_image(item["image_url"]["url"])
+                            message_content.append(
+                                {
+                                    "type": "image",
+                                    "source": source,
+                                }
+                            )
+                        else:
+                            message_content.append(item)
+                    else:
+                        raise ValueError(
+                            f"""Content items 
+                                must be str or dict, instead was: {type(item)}"""
+                        )
+            else:
+                message_content = message.content
+
+            formatted_messages.append(
+                {
+                    "role": role,
+                    "content": message_content,
+                }
+            )
     return system, formatted_messages
```

Comment:
A lot of this can be improved by early returns and refactoring to separate functions. I have not tested the validity of this change, so do check against existing logic before taking these changes as-is. Adding some unit tests to check the validity of these would be even better.

```python
def _handle_anthropic_system_message(message):
    if not isinstance(message.content, str):
        raise ValueError(
            "System message must be a string, "
            f"instead was: {type(message.content)}"
        )
    return message.content, _message_type_lookups["human"]

def _handle_anthropic_message(message):
    role = _message_type_lookups[message.type]
    if isinstance(message.content, str):
        return message.content, role
    
    assert isinstance(message.content, list), "Anthropic message content must be str or list of dicts"
    
    content = []
    for item in message.content:
        if isinstance(item, str):
            content.append({"type": "text", "text": item})
        elif isinstance(item, dict):
            if "type" not in item:
                raise ValueError("Dict content item must have a type key")
            if item["type"] == "image_url":
                source = _format_image(item["image_url"]["url"])
                content.append({"type": "image", "source": source})
            else:
                content.append(item)
        else:
            raise ValueError(f"Content items must be str or dict, instead was: {type(item)}")
    return content, role

def _format_anthropic_messages(
    messages: List[BaseMessage],
) -> Tuple[Optional[str], List[Dict]]:
    """Format messages for anthropic."""
    """
    [
        {
            "role": _message_type_lookups[m.type],
            "content": [_AnthropicMessageContent(text=m.content).dict()],
        }
        for m in messages
    ]
    """    
    if isinstance(messages, langchain_system.BaseMessage):
        if messages.type == "system":
            system, role = _handle_anthropic_system_message(messages)
        else:
            content, role = _handle_anthropic_message(messages)
    return system, [{"role": role, "content": content}]
    
    system = None
    formatted_messages = []
    for i, message in enumerate(messages):
        if message.type == "system":
            if i != 0:
                raise ValueError("System message must be at beginning of message list.")
            system, _ = _handle_anthropic_system_message(message)
            continue
        content, role = _handle_anthropic_message(message)
        formatted_messages.append({"role": role, "content": content})
    return system, formatted_messages
```

Reply:
Implemented and test.